File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int carry = 0;
        ListNode dummy = new ListNode(0);
        ListNode current = dummy;
        while (!((l1 == null && l2 == null) && (carry == 0))) {// l1와 l2 둘다 null이고 carry가 0이면 종료. 그게 아니라면 계속

            int val1 = l1 == null ? 0 : l1.val;
            int val2 = l2 == null ? 0 : l2.val;

            int sum = val1 + val2 + carry;
            carry = sum / 10;
            int real = sum % 10;

            ListNode node = new ListNode(real);
            current.next = node;
            current = node;

            l1 = l1 == null ? null : l1.next;
            l2 = l2 == null ? null : l2.next;
        }
        return dummy.next;

    }

}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carry = 0
        iterator = ListNode()
        ans = iterator
        while (l1 or l2) or carry != 0: #l1 l2둘중 하나라도 null이 아니면 안끝남. carry도 0아니면 안끝남
            val1 = l1.val if l1 else 0
            val2 = l2.val if l2 else 0

            sum = val1 + val2 + carry
            valSum = sum % 10
            carry = sum // 10

            #print(f"val1={val1},val2={val2}, sum={sum}, valSum={valSum}, carry={carry}")
            node = ListNode(valSum)
            iterator.next = node
            iterator = node

            l1 = l1.next if l1 and l1.next else None
            l2 = l2.next if l2 and l2.next else None
        
        return ans.next